sponsors/forms.py

```diff
@@ -463,3 +463,55 @@ def get_notification(self):
             subject=self.cleaned_data["subject"],
         )
         return self.cleaned_data.get("notification") or default_notification
+
+
+class SponsorUpdateForm(forms.ModelForm):
+    READONLY_FIELDS = [
+        "name",
+    ]
+
+    def __init__(self, *args, **kwargs):
+        super().__init__(*args, **kwargs)
+        formset_kwargs = {"prefix": "contact", "instance": self.instance}
+        factory = forms.inlineformset_factory(
+            Sponsor,
+            SponsorContact,
+            form=SponsorContactForm,
+            extra=0,
+            min_num=1,
+            validate_min=True,
+            can_delete=True,
+            can_order=False,
+            max_num=5,
+        )
+        if self.data:
+            self.contacts_formset = factory(self.data, **formset_kwargs)
+        else:
+            self.contacts_formset = factory(**formset_kwargs)
+        # display fields as read-only
+        for disabled in self.READONLY_FIELDS:
+            self.fields[disabled].widget.attrs['readonly'] = True
+
+    class Meta:
+        exclude = ["created", "updated", "creator", "last_modified_by", "web_logo", "print_logo"]
```

Comment:
I think we can include web_logo and print_logo. But should not allow them to be cleared, only replaced.